src/GovUk.Education.ExploreEducationStatistics.Public.Data.Api.Tests/Controllers/DataSetsControllerTests.cs

```diff
@@ -714,440 +719,859 @@ public GetDataSetMetaTests(TestApplicationFactory testApp) : base(testApp)
         {
-            DataSet dataSet = DataFixture
-                .DefaultDataSet()
-                .WithStatusPublished();
+            public TypesQueryParameterTests(TestApplicationFactory testApp) : base(testApp)
+            {
+            }
 
-            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+            [Fact]
+            public async Task TypesNotSpecified_ReturnsAllMetadata()
+            {
+                DataSet dataSet = DataFixture
+                    .DefaultDataSet()
+                    .WithStatusPublished();
+
+                await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+                DataSetVersion dataSetVersion = DataFixture
+                    .DefaultDataSetVersion(
+                        filters: 1,
+                        indicators: 1,
+                        locations: 1,
+                        timePeriods: 2)
+                    .WithStatusPublished()
+                    .WithDataSetId(dataSet.Id)
+                    .FinishWith(dsv => dataSet.LatestVersion = dsv);
+
+                await TestApp.AddTestData<PublicDataDbContext>(context =>
+                {
+                    context.DataSetVersions.Add(dataSetVersion);
+                    context.DataSets.Update(dataSet);
+                });
 
-            DataSetVersion dataSetVersion = DataFixture
-                .DefaultDataSetVersion(
-                    filters: 1,
-                    indicators: 1,
-                    locations: 1,
-                    timePeriods: 3)
-                .WithStatus(dataSetVersionStatus)
-                .WithDataSetId(dataSet.Id);
+                var response = await GetDataSetMeta(dataSet.Id);
 
-            await TestApp.AddTestData<PublicDataDbContext>(context => context.DataSetVersions.Add(dataSetVersion));
+                var content = response.AssertOk<DataSetMetaViewModel>(useSystemJson: true);
 
-            var response = await GetDataSetMeta(dataSet.Id, dataSetVersion.Version);
+                Assert.NotNull(content);
 
-            response.AssertForbidden();
-        }
+                Assert.NotEmpty(content.Filters);
+                Assert.NotEmpty(content.Locations);
+                Assert.NotEmpty(content.Indicators);
+                Assert.NotEmpty(content.TimePeriods);
+            }
 
-        [Fact]
-        public async Task VersionExistsForOtherDataSet_Returns404()
-        {
-            DataSet dataSet1 = DataFixture
-                .DefaultDataSet()
-                .WithStatusPublished();
+            [Theory]
+            [InlineData(MetadataType.Filters)]
+            [InlineData(MetadataType.Locations)]
+            [InlineData(MetadataType.Indicators)]
+            [InlineData(MetadataType.TimePeriods)]
+            public async Task OneTypeSpecified_OnlyReturnsMetadataForSpecifiedType(MetadataType metadataType)
```

Comment:
Not a big deal, but could shorten all these test case names with `_ReturnsOnlySpecifiedMetaType`.

Similar thing could apply in test cases below as well